conformance/lang/expressions/literals/floating_point_literal.balt

```diff
@@ -0,0 +1,6092 @@
+
+    d = 4.94065645841246544E-324;
+    // io:println(d); // @output 4.9E-324
+
+    d = 4.94065645841246544E-325;
+    io:println(d); // @output 0.0
+
+    d = 0x1.fffffffffffffp+1023;
+    io:println(d); // @output 1.7976931348623157E308 
+
+    d = 0x1.fffffffffffffp+1024;
+    io:println(d); // @output Infinity 
+
+    d = 0x1.ffffffffffffffp+1023;
+    io:println(d); // @output Infinity 
+
+    d = 0x1.ffffffffffffffp+1024;
+    io:println(d); // @output Infinity 
+
+    d = 0x1p-1074;
+    io:println(d); // @output 4.9E-324 
+
+    d = 0x1p-1075;
+    io:println(d); // @output 0.0 
+
+    d = 0x0p-1074;
+    io:println(d); // @output 0.0 
+
+    d = 0x0p-1075;
+    io:println(d); // @output 0.0 
+}
+
+Test-Case: output
+Description: Test floating point literal max and min with decimal type descriptor.
+Labels: decimal 
+
+function init() {
+    decimal e = 1E-6176;
+    io:println(e); // @output 1E-6176
+
+    e = 9.99E+6111;
+    io:println(e); // @output 9.99E+6111
+}
+
+Test-Case: output
+Description: Test floating point literal max and min with floating point type descriptor.
+Labels: decimal, float, HexFloatingPointLiteral, FloatTypeSuffix, DecimalTypeSuffix
+
+function init() {
+    float|decimal f = 1.7976931348623157E+308;
+    io:println(f); // @output 1.7976931348623157E308
+
+    f = 1.7976931348623157E+308F;
+    io:println(f); // @output 1.7976931348623157E308
+
+    f = 1.7976931348623157E+309;
+    io:println(f); // @output Infinity
+
+    f = 1.7976931348623157E+309f;
+    io:println(f); // @output Infinity
+
+    f = 4.94065645841246544E-324;
+    io:println(f); // @output 4.9E-324
+
+    f = 4.94065645841246544E-324F;
+    io:println(f); // @output 4.9E-324
+
+    f = 4.94065645841246544E-325;
+    io:println(f); // @output 0.0
+
+    f = 4.94065645841246544E-325f;
+    io:println(f); // @output 0.0
+
+    f = 0x1.fffffffffffffp+1023;
+    io:println(f); // @output 1.7976931348623157E308 
+
+    f = 0x1.fffffffffffffp+1024;
+    io:println(f); // @output Infinity 
+
+    f = 0x1.ffffffffffffffp+1023;
+    io:println(f); // @output Infinity 
+
+    f = 0x1.ffffffffffffffp+1024;
+    io:println(f); // @output Infinity 
+
+    f = 0x1p-1074;
+    io:println(f); // @output 4.9E-324 
+
+    f = 0x1p-1075;
+    io:println(f); // @output 0.0 
+
+    f = 0x0p-1074;
+    io:println(f); // @output 0.0 
+
+    f = 0x0p-1075;
+    io:println(f); // @output 0.0 
+
+    f = 1E-6176;
+    io:println(f); // @output 0.0
+
+    f = 9.99E+6111;
+    io:println(f); // @output Infinity
+
+    f = 1E-6176d;
+    io:println(f); // @output 1E-6176
+
+    f = 9.99E+6111D;
+    io:println(f); // @output 9.99E+6111
+}
+
+Test-Case: error
+Description: Test decimal floating point numbers with leading zeros
+Labels: decimal, float, FloatTypeSuffix, DecimalTypeSuffix
+
+function init() {
+    float|decimal a = 03f; // @error invalid floating point with leading '0'
+
+    a = 03F; // @error invalid floating point with leading '0'
```

Comment:
we can use `_` in these cases where the reference is anyway ignored

Reply:
Check other plcaes as well

Reply:
[808331d](https://github.com/ballerina-platform/ballerina-spec/pull/947/commits/808331da7e85b86e4b950f73bcddde9b67a754dc)